tests_e2e/test_app.py

```diff
@@ -0,0 +1,58 @@
+import os
+import pytest
+import requests
+import time
+from dotenv import find_dotenv, load_dotenv
+from selenium import webdriver
+from threading import Thread
+from todo_app.app import create_app
+from todo_app.trello_api import create_board
+
+board_id = ''
+
+def create_trello_board():
+    url = 'https://api.trello.com/1/boards/'
+    query = { 'key': os.environ.get('TRELLO_API_KEY'), 'token': os.environ.get('TRELLO_API_TOKEN'), 'name': os.environ.get('BOARD_NAME') }
+    response = requests.post(url, params=query)
+    result = response.json()
+    return result['id']
+
+def delete_trello_board():
+    url = f'https://api.trello.com/1/boards/{board_id}'
+    query = { 'key': os.environ.get('TRELLO_API_KEY'), 'token': os.environ.get('TRELLO_API_TOKEN') }
+    requests.delete(url, params=query)
+
+@pytest.fixture(scope='module')
+def app_with_temp_board():
+    file_path = find_dotenv('.env')
+    load_dotenv(file_path, override=False)
+    os.environ['BOARD_NAME'] = 'Test Board'
+    global board_id
+    board_id = create_trello_board()
+    os.environ['TRELLO_BOARD_ID'] = board_id
+    application = create_app()
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield application
+    # Tear Down
+    thread.join(1)
+    delete_trello_board()
+
+@pytest.fixture(scope='module')
+def driver():
+    current_dir = os.getcwd()
+    chromedriver_executable = f'{current_dir}/chromedriver'
+    with webdriver.Chrome(chromedriver_executable) as driver:
+        yield driver
+
+def test_task_journey(driver, app_with_temp_board):
+    driver.get('http://localhost:5000/')
+    assert driver.title == 'To-Do App'
+    time.sleep(2)
```

Comment:
Sleeps should be a last resort as it makes the test take longer than it needs to

Most of the time you can write your test to look for an element. And you can set `driver.implicitly_wait(2)` at the start of the test to make it wait **up to** 2 seconds when trying any of the `driver.find_element_by...` functions for the element to appear.